src/plugins/workspace/public/utils.ts

```diff
@@ -202,3 +203,44 @@ export const getDataSourcesList = (client: SavedObjectsStart['client'], workspac
       }
     });
 };
+
+export const convertNavGroupToWorkspaceUseCase = ({
+  id,
+  title,
+  description,
+  navLinks,
+  type,
+  order,
+}: NavGroupItemInMap): WorkspaceUseCase => ({
+  id,
+  title,
+  description,
+  features: navLinks.map((item) => item.id),
+  systematic: type === NavGroupType.SYSTEM,
+  order,
+});
+
+export const isEqualWorkspaceUseCase = (a: WorkspaceUseCase, b: WorkspaceUseCase) => {
```

Comment:
Do we need to check if the use case from constant equal to the one from navGroup? I thought match on id should be good enough.

Reply:
For now, the `isEqualWorkspaceUseCase` only been used in `UseCaseService`. It's just for avoid duplicate changes fired. The `appUpdater$` in `filterNavLinks` will generate new nav links. Then the `getNavGroupsMap$` will emit new nav groups map. It will generate an infinite loop. So we add `distinctUntilChanged` after `getNavGroupsMap$` to avoid these.

Reply:
I c, thanks.